adminSiteClient/ChartEditorPage.tsx

```diff
@@ -278,12 +289,71 @@ export class ChartEditorPage
         return grapherConfigDetails
     }
 
-    @computed get invalidDetailReferences() {
-        const { subtitle, note } = this.currentDetailReferences
-        return {
-            subtitle: subtitle.filter((key) => !this.details[key]),
-            note: note.filter((key) => !this.details[key]),
-        }
+    @computed
+    get invalidDetailReferences(): ChartEditorManager["invalidDetailReferences"] {
+        const { currentDetailReferences } = this
+        const keys = getIndexableKeys(currentDetailReferences)
+
+        const invalidDetailReferences = Object.fromEntries(
+            keys.map((key: FieldWithDetailReferences) => [
+                key,
+                currentDetailReferences[key].filter(
+                    (term: string) => !this.details[term]
+                ),
+            ])
+        ) as DetailReferences
+
+        return invalidDetailReferences
+    }
+
+    @computed get errorMessages(): ChartEditorManager["errorMessages"] {
+        const { invalidDetailReferences } = this
+
+        const errorMessages: ChartEditorManager["errorMessages"] = {}
+
+        // add error messages for each field with invalid detail references
+        getIndexableKeys(invalidDetailReferences).forEach(
+            (key: FieldWithDetailReferences) => {
+                const references = invalidDetailReferences[key]
+                if (references.length) {
+                    errorMessages[
+                        key
+                    ] = `Invalid detail(s) specified: ${references.join(", ")}`
+                }
+            }
+        )
+
+        return errorMessages
+    }
+
+    @computed
+    get errorMessagesForDimensions(): ChartEditorManager["errorMessagesForDimensions"] {
```

Comment:
Yes, indicator names are not only used as default axis labels, they're used across our UI. If we allowed DoD syntax here, the raw DoD syntax string would show up in all kinds of places (e.g. the table). That's true for most of our text properties that don't support DoDs but what makes this special is that adding detail syntax to an indicator name would appear to work for axis labels, i.e. an axis label would actually render a DoD, but it would be broken in the sense that we don't keep track of DoDs in indicator names, thus the DoD wouldn't be appended to static charts, for example.